### DISCUSSION TITLE
How to add variations to the listing

### DISCUSSION THREAD

**Discussion by dusariudaykumar** [amzn/selling-partner-api-models#3724] [General]
Hello there,

 Is there any way to add variations to the listing while creating itself?  I was trying to create a listing for a t-shirt using listings API [putListingsItem](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#put-listings2021-08-01itemsselleridsku) but I couldn't find any resource on how to add variations to it.
 @leonardonakagawa 

Thanks

**Comment by vaidas-lungis** [comment]
Documentation
https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#configuring-variation-families

Items in this repository:

https://github.com/amzn/selling-partner-api-docs/discussions/3300
https://github.com/amzn/selling-partner-api-docs/issues/3353

Others in the wild

https://github.com/jlevers/selling-partner-api/issues/472

**Reply by wszwc** [reply]
Hello, thank you for your help, I took a cursory look and what I don't understand is whether it is possible to upload the main variant and all sub-variants of a product in one go. If so, what is the json structure? The interface parameters seem to only support one go. Upload a variation

**Comment by leonardonakagawa** [comment]
Hi @wszwc,

If you are creating the items using putListingsItem (1x1) you cannot create the parent (main) and child (sub-variants) in a single go. For that, you can rely on the JSON_LISTINGS_FEED using the same JSON structure from 1x1 listings API.

You can rely on this documentation, [Creating and updating listings items in bulk](https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#creating-and-updating-listings-items-in-bulk) to perform this action.


If the information available in the documentation link provided above does not fully resolve your inquiry, please [open a support case](https://developer.amazonservices.com/support) with us.

Thanks,
Leonardo Nakagawa
Solutions Architect, Selling Partner API Developer Support

**Reply by wszwc** [reply]
Hello, I am using JSON_LISTINGS_FEED to batch upload products according to your suggestion, but I encountered a new problem. After I uploaded many products, it became extremely difficult to determine whether they were successfully uploaded. The results returned by the feed report do not seem to fully represent their success.

**Reply by wszwc** [reply]
There is also a very important question. How do I determine the variant themes and corresponding attributes? The json product type template gives me up to 60 variant themes, but I see in the Amazon seller backend that this product type only has three variant themes. , I didn’t see how he found out the attributes corresponding to the variant theme.
